pkg/api/filter/filter.go

```diff
@@ -0,0 +1,33 @@
+package filter
+
+import "net/http"
+
+// Filter is a wrapper around a http.HandlerFunc.
+type Filter interface {
+	Filter(http.HandlerFunc) http.HandlerFunc
+}
+
+type chain struct {
+	filters []Filter
+}
+
+// NewFilterChain returns a Filter capable of applying
+// the provided Filter(s) to another http.HandlerFunc
+func NewFilterChain(filterChain []Filter) Filter {
+	return chain{
+		filters: filterChain,
+	}
+}
+
+// Filter applies a chain of FilterFunction to a target http.HandlerFun
+// and returns a new http.HandlerFunc that wraps the chain
+func (h chain) Filter(
```

Comment:
`Filter` sound like a verb or an imperative to actually _do_ the filtering, but this function really just returns a handler. The handler is a decorated one that _does_ the actual filtering. Maybe this function is better renamed as something like `GetHandler(...)`?

Reply:
No problem.